app/global/MainModule.java

```diff
@@ -228,14 +228,18 @@ protected void configure() {
     @SuppressFBWarnings("UPM_UNCALLED_PRIVATE_METHOD") // Invoked reflectively by Guice
     private AlertExecutionContext provideAlertExecutionContext(
             final Config config,
-            final QueryExecutor executor
+            final QueryExecutor executor,
+            final PeriodicMetrics metrics
     ) {
         final FiniteDuration interval = ConfigurationHelper.getFiniteDuration(config, "alerting.execution.defaultInterval");
         final java.time.Duration queryOffset = ConfigurationHelper.getJavaDuration(config, "alerting.execution.queryOffset");
 
         final Schedule defaultAlertSchedule = new UnboundedPeriodicSchedule.Builder()
                 .setPeriod(TimeAdapters.toChronoUnit(interval.unit()))
                 .setPeriodCount(interval.length())
+                .setOverrunReporter(overrunPeriodCount -> {
+                    metrics.recordCounter("jobs/executor/overrunPeriods", overrunPeriodCount);
```

Comment:
I couldn't encode anything about the job since this isn't within the executor framework at this point. This provider is for alerts and so I *could* mark the metric as alert-specific but I figure that could be messy since it would couple it to the `alert_job` type at a distance.

Perhaps the better thing to do here is refactor this into `AlertExecutionContext`, where the schedule is constructed per-alert, and somehow plumb this in.

Reply:
I would be okay with logging a warning or error, ideally with an exception